cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/store/AppMetadataStoreTest.java

```diff
@@ -1480,16 +1481,75 @@ public void testConcurrentCreateAppAfterTheFirstVersion() throws Exception {
     Assert.assertEquals(1 + numThreads, appEditNumber.get());
   }
 
+  @Test
+  public void testDeleteRunsStartedBefore() throws Exception {
+    ProgramId programId = NamespaceId.DEFAULT.app("test").workflow("test");
+
+    Instant ttlCutoff = Instant.ofEpochSecond(1631629389);
+
+    ProgramRunId shouldExpire = createCompletedRun(programId, ttlCutoff.minusSeconds(10));
```

Comment:
Would recommend bigger data size for unit tests, like writing a hundred run records with various versions/program types/appNames and verify the results after janitor delete

Reply:
Updated to create 100 to expire and 100 to keep, distributed between 15 different app:workflow combinations.